harness/determined/exec/harness.py

```diff
@@ -75,9 +75,10 @@ def main(chief_ip: Optional[str]) -> int:
     )
 
     multi_machine_trial = len(info.container_addrs) > 1
-    hvd_config = horovod.HorovodContext.from_configs(
-        env.experiment_config, env.hparams, multi_machine_trial
-    )
+    multi_slot_trial = (env.experiment_config.slots_per_trial() > 1
+                        and not env.experiment_config.native_parallel_enabled())
+
+    distributed_backend = "horovod" if multi_machine_trial or multi_slot_trial else None
```

Comment:
This looks like clear evidence that you could use `if context.distributed.size > 1` everywhere that `if hvd_config.use` is used